### ISSUE TITLE
Using with images having varying number of segments

### ISSUE THREAD

**Issue by somnathrakshit** [meetshah1995/pytorch-semseg#108]
I have a dataset of images containing varying number of segments. I also have pixel wise  labels for each of them. How can I use this dataset with this library?

**Comment by raj-shah** [comment]
If you have your own/custom dataset with ground truth (pixel-wise labels) available, it's just a matter of writing your own train and validation dataloaders. You should then be able to integrate it seamlessly with this repo.

**Comment by somnathrakshit** [comment]
Thanks. One quick question. Not all images have the same number of clusters. Will that cause any issue?

**Comment by raj-shah** [comment]
By clusters, I am assuming you mean number of classes present in an image? If so, then no. I would suggest looking into the datasets commonly used in semantic segmentation.

**Comment by somnathrakshit** [comment]
`--dataset        Dataset to use ['pascal, camvid, ade20k etc']`

In my case with custom datasets, is this parameter required while training?

**Comment by raj-shah** [comment]
Depends on how you implement it. If you are looking to do the same as done in this repo, then you need to modify your .json file to point to your dataset's location, and write the necessary loader scripts for train/validation routines. Look at `__init__.py` under `loader`.

Re code samples, I think there are quite a few repos on github where people have used these architectures on custom datasets.

**Comment by meetshah1995** [comment]
I plan to add a base class for dataset in the update soon, so that you can inherit that class and create your own dataset iterator. Closing this for now?
